merge_test.go

```diff
@@ -201,6 +201,110 @@ func TestShouldKeyMergeArrayOfHashes(t *testing.T) {
 	})
 }
 
+func TestShouldInsertIntoArrayBasedOnIndex(t *testing.T) {
+	Convey("We should insert into arrays based on index", t, func() {
+		Convey("If insert token with after and index is found", func() {
+			result, idx := shouldInsertIntoArrayBasedOnIndex([]interface{}{"(( insert after 0 ))", "stuff"})
+			So(result, ShouldBeTrue)
+			So(idx, ShouldEqual, 1)
+		})
+
+		Convey("If insert token with before and index is found", func() {
+			result, idx := shouldInsertIntoArrayBasedOnIndex([]interface{}{"(( insert before 0 ))", "stuff"})
+			So(result, ShouldBeTrue)
+			So(idx, ShouldEqual, 0)
+		})
+
+		Convey("If insert token with after and index is found independent of missing whitespaces", func() {
+			result, idx := shouldInsertIntoArrayBasedOnIndex([]interface{}{"((insert after 0))", "stuff"})
+			So(result, ShouldBeTrue)
+			So(idx, ShouldEqual, 1)
+		})
+
+		Convey("If insert token with after and index is found with a lot of additional whitespaces", func() {
+			result, idx := shouldInsertIntoArrayBasedOnIndex([]interface{}{"((  insert   after   0  ))", "stuff"})
+			So(result, ShouldBeTrue)
+			So(idx, ShouldEqual, 1)
+		})
+
+		Convey("But not if index is obviously out of bounds", func() {
+			result, idx := shouldInsertIntoArrayBasedOnIndex([]interface{}{"(( insert before -1 ))", "stuff"})
+			So(result, ShouldBeFalse)
+			So(idx, ShouldEqual, -1)
+		})
```

Comment:
Can you add a similar test here for when the index is out of bounds in the other direction?